test/analysis/buildTypeWeights.test.ts

```diff
@@ -121,14 +126,183 @@ xdescribe('Test buildTypeWeightsFromSchema function', () => {
             });
         });
 
+        test('types with arguments', () => {
+            schema = buildSchema(`
+                type Query {
+                    character(id: ID!): Character
+                }
+                type Character {
+                    id: ID!
+                    name: String!
+                }`);
+            expect(buildTypeWeightsFromSchema(schema)).toEqual({
+                Query: {
+                    weight: 1,
+                    fields: {},
+                },
+                Character: {
+                    weight: 1,
+                    fields: {
+                        id: 0,
+                        name: 0,
+                    },
+                },
+            });
+        });
+
+        test('enum types', () => {
+            schema = buildSchema(`
+                type Query {
+                    hero(episode: Episode): Character
+                }
+                type Character {
+                    id: ID!
+                    name: String!
+                }
+                enum Episode {
+                    NEWHOPE
+                    EMPIRE
+                    JEDI
+                }`);
+            expect(buildTypeWeightsFromSchema(schema)).toEqual({
+                Query: {
+                    weight: 1,
+                    fields: {},
+                },
+                Character: {
+                    weight: 1,
+                    fields: {
+                        id: 0,
+                        name: 0,
+                    },
+                },
+                Episode: {
+                    weight: 0,
+                    fields: {},
+                },
+            });
+        });
+
+        // ? varibale weight
+        test('fields returning lists of objects', () => {
+            schema = buildSchema(`
+                type Query {
+                    reviews(episode: Episode!, first: Int): [Review]
+                }
+                type Review {
+                    episode: Episode
+                    stars: Int!
+                    commentary: String
+                }
+                enum Episode {
+                    NEWHOPE
+                    EMPIRE
+                    JEDI
+                }`);
+            expect(buildTypeWeightsFromSchema(schema)).toEqual({
+                Query: {
```

Comment:
The end goal is to have a function that can be used to calculate the total type complexity for a field by multiplying the arg value times the weight of the Resolved Type. 

For example, if we receive a query `reviews(episode: 'NEWHOPE', first: 5)`, the type complexity algorithm would call the function with the value of `first` the weight of a `Review` to get 5 * 1 = 5; 

This function could be global that receives both arguments or a unique function could be assigned to each applicable field as below. 

When looking up a field weight, we would check the type of the value. If the weight is a number, simply return the number otherwise call the associated function with relevant args. 

We will also need to determine if the relevant information (variable names and resolve types) is available in the `GraphQLSchemaObject` or the AST created by parsing the schema object

`first`, `last`, `limit` are conventional keywords for limiting results. Later on we could make these keywords configurable to provide a more unopinionated solution. 

````javascript
Query {
  weight: 1,
  fields: {
    reviews: (type) => args[multiplierName] * typeWeightObject[type].weight
  },
}, 

Review {
  weight: 1,
  fields: {
    stars: 0,
    commentary: 0,  
  }
}
````

- [ ] Create an issue or add these comments to the appropriate existing issue

Reply:
I've added a function definition similar to this one to the expected type weight object wherever its necessary (other than the unknown list sizes which we are putting off). I've also added fixme to update the test based on what we decide to do in implementation.

Reply:
Also thinking we could implement a closure around the type when setting the `reviews` function definition so it only takes in the argument from `first` when calculating the complexity.